### ISSUE TITLE
Publish to Maven Central too(?)

### ISSUE THREAD

**Issue by ksclarke** [eXist-db/mvn-repo#6]
It's great to have this available, but it would be wonderful to also have these artifacts in Maven Central. Since Maven discourages publishing artifacts in the Maven Central repo that have dependencies on external Maven repositories, not having these artifacts in Central (in addition to here) makes it less convenient for people who would like to publish related projects through Maven.  So, long story short, I was wondering whether you'd consider publishing to the Maven Central too.  Thanks for considering!

**Comment by adamretter** [comment]
Hi @ksclarke indeed it would be a great thing to do. At the moment the process of creating and publishing those jars to the mvn-repo is manual. We would need to automate this ideally if we want to go further and publish to Maven Central. Perhaps something you could help with?

**Comment by ksclarke** [comment]
Yes, I'd be glad to help out with automating this.  I'll start looking at it and get back to you. Thanks.

**Comment by shabanovd** [comment]
I have automatic script that create maven artifacts out of eXist source. If others agree I can cron it on my server and do publishing to central mvn-repo including "current" develop branch as snapshot.

**Comment by shabanovd** [comment]
here the source if interesting https://github.com/shabanovd/utils-source-maker

**Comment by adamretter** [comment]
Thanks @shabanovd however I think it should be added as a series of Ant targets to eXists existing build.

**Comment by adamretter** [comment]
@ksclarke Could you do it as some additional Ant targets in eXists build/

**Comment by shabanovd** [comment]
It do

On Thu, Feb 26, 2015 at 9:07 PM, Adam Retter <notifications@github.com>
wrote:

> Thanks @shabanovd <https://github.com/shabanovd> however I think it
> should be added as a series of Ant targets to eXists existing build.
>

-- 
Dmitriy Shabanov

**Comment by ksclarke** [comment]
Not having looked at it yet, my first thought was to call the ant tasks from a Maven build using the Maven Ant Tasks plugin.  Then Maven uses Ant to do the existing builds but uses it's native methods for pushing the builds into the central repo.

The alternative though, if you wanted to stay completely in the Ant world, would to be to use Ivy with Ant to publish the artifacts.  I don't have experience with this, but would be glad to look into it if you'd prefer not to add Maven into the mix.

**Comment by dizzzz** [comment]
We have ivy already included

**Comment by ksclarke** [comment]
Ah, great. Like I said, haven't looked into it yet (this would be an evening project for me).  As I understand it, then, Ivy can be configured to publish to Maven central (through Sonatype... do you all have a Sonatype account?) so perhaps that's the best path to investigate.

**Comment by dizzzz** [comment]
I do have à sonatype account....

**Comment by adamretter** [comment]
Yes I have access to Sonatype with my own account. We do plan to move to Maven in the future, but that is a longer term plan. Cheers

**Comment by ksclarke** [comment]
Okay, will proceed with Ivy then using my Sonatype and, then, if I get something usable one of you should be able to just swap in your info and actually do the push. Thanks.

**Comment by adamretter** [comment]
@ksclarke Did you make any progress?

**Comment by ksclarke** [comment]
After writing a basic POM I started looking at the dependencies that weren't in Maven Central. I thought perhaps working towards moving the dependencies into Maven Central would be the best way to resolve the issue. I started on the webdav dependency, but then got swamped with other work and never completed it. It's still something I'm interested in, but I don't see that I'll be able to have the spare cycles any time soon to work on it.
